### DISCUSSION TITLE
Changing base branch triggers GitHub to dismiss reviews

### DISCUSSION THREAD

**Discussion by kenocrates** [community/community#45921] [Pull Requests] [answered]
### Select Topic Area

Bug

### Body

Until recently (as far back as 11 Jan 2023 or as recently as 29 Jan 2023) changing the base branch of a PR did not automatically dismiss previous accepted reviews as stale. Now it does, with the comment "The base branch was changed".

**Comment by nhurden** [comment]
This seems to break `git spr merge`, since that relies on changing the base branch to rebase-merge a stack of PRs: https://github.com/ejoffe/spr

**Reply by kenocrates** [reply]
Not an answer but perhaps this will lean support for reverting the change.

**Comment by kenocrates** [comment, marked as answer]
I have observed today that the behaviour has reverted.

**Reply by joehorsnell** [reply]
This discussion should not be marked as answered - it seems this issue was later reintroduced (see #57513 and #57944) and has been broken now for nearly two and a half years. Why is this taking so long to address GitHub?

**Comment by edenman** [comment]
<img width="708" alt="Screenshot 2024-04-16 at 15 38 57" src="https://github.com/community/community/assets/1063557/8884914f-ac77-4fe2-a860-743dcb5efdd8">

I'm seeing this.  Can't find any setting that seems like it would control this behavior, but maybe it's "Require approval of the most recent reviewable push" and changing the base branch counts as a reviewable push for some reason?

**Comment by jasonhuhx** [comment]
We are still seeing this issue in stacked PR workflows.

Scenario:
1. PR A is the base for PR B.
2. PR A is merged.
3. GitHub automatically retargets PR B to `master`.
4. PR B’s existing approvals are dismissed with the base-change event.

Our repository setting to dismiss approvals when the base branch changes is turned off, so these approvals should be preserved. This is especially disruptive for stacked PR workflows because the base retargeting is expected after the lower PR merges, and no new commits were pushed to the top PR.

This appears to be the same bug described here and seems to have been reintroduced or never fully resolved. It would be helpful if GitHub could clarify whether automatic retargeting after a stacked PR merge is intended to bypass the repository setting, or treat this as a bug and preserve approvals as configured.
